src/object/schema_manager.c

```diff
@@ -3189,6 +3269,24 @@ sm_is_system_class (MOP op)
   return sm_get_class_flag (op, SM_CLASSFLAG_SYSTEM);
 }
 
+static int
+system_class_def_compare (const void *a, const void *b)
+{
+  int a_len = ((const SYSTEM_CLASS_DEF *) a)->len;
+  int b_len = ((const SYSTEM_CLASS_DEF *) b)->len;
+
+  if (a_len < b_len)
+    {
+      return -1;
+    }
+  else if (a_len > b_len)
+    {
+      return 1;
+    }
+
+  return 0;
+}
```

Comment:
Please make the function as below.
```
static int
system_class_def_compare (const void *a, const void *b)
{
    const SYSTEM_CLASS_DEF * sa = (const SYSTEM_CLASS_DEF *) a;
    const SYSTEM_CLASS_DEF * sb = (const SYSTEM_CLASS_DEF *) b;

    if( sa->len != sb->len)
    {
        return sa->len - sb->len;
    }
    return strcmp(sa->name, sb->name);
}
```

That way you can make the bottom for loop more efficient like this:
```
int cmp;
for (i = 0; i < count; i++)
    {
        cmp = system_class_def_compare(downcase_name, system_classes[i].name) ;
        if(cmp <= 0)
          {
              return  (cmp == 0);
           }        
    }
```